pkg/agent/controller/reconciliation_test.go

```diff
@@ -102,26 +111,33 @@ var _ = Describe("Reconciliation", func() {
 			testutil.EnsureNoActionsForResource(&t.cluster1.localDynClient.Fake, "endpointslices", "delete")
 
 			brokerDynClient := t.syncerConfig.BrokerClient.(*fake.FakeDynamicClient)
-			testutil.EnsureNoActionsForResource(&brokerDynClient.Fake, "serviceimports", "delete")
 			testutil.EnsureNoActionsForResource(&brokerDynClient.Fake, "endpointslices", "delete")
+
+			// For migration cleanup, it may attempt to delete a local legacy ServiceImport from the broker so ignore it.
+			Consistently(func() bool {
+				siActions := brokerDynClient.Fake.Actions()
+				for i := range siActions {
+					if siActions[i].GetResource().Resource == "serviceimports" && siActions[i].GetVerb() == "delete" &&
+						!strings.Contains(siActions[i].(testing.DeleteAction).GetName(), t.cluster1.clusterID) {
+						fmt.Printf("****DEL: %s, %s, %s\n", siActions[i].GetResource().Resource, siActions[i].GetVerb(),
```

Comment:
Do we want to keep this?

Reply:
oh no - nice catch - not sure how I missed that.